conftest.py

```diff
@@ -32,3 +39,37 @@ def sample_dag():
     with create_session() as session:
         session.query(DagRun).delete()
         session.query(TI).delete()
+
+
+@pytest.fixture
+def tmp_table(sql_server):
+    sql_name, hook = sql_server
+
+    if isinstance(hook, SnowflakeHook):
+        return TempTable(
+            conn_id=hook.snowflake_conn_id,
+            database=hook.database,
+            warehouse=hook.warehouse,
+        )
+    elif isinstance(hook, PostgresHook):
+        return TempTable(conn_id=hook.postgres_conn_id, database=hook.schema)
+    elif isinstance(hook, SqliteHook):
+        return TempTable(conn_id=hook.sqlite_conn_id, database="sqlite")
+    elif isinstance(hook, BigQueryHook):
+        return TempTable(conn_id=hook.gcp_conn_id)
+
+
+@pytest.fixture
+def sql_server(request):
+    sql_name = request.param
+    hook_parameters = test_utils.SQL_SERVER_HOOK_PARAMETERS.get(sql_name)
+    hook_class = test_utils.SQL_SERVER_HOOK_CLASS.get(sql_name)
+    if hook_parameters is None or hook_class is None:
+        raise ValueError(f"Unsupported SQL server {sql_name}")
+    hook = hook_class(**hook_parameters)
+    schema = hook_parameters.get("schema", test_utils.DEFAULT_SCHEMA)
+    if not isinstance(hook, BigQueryHook):
+        hook.run(f"DROP TABLE IF EXISTS {schema}.{OUTPUT_TABLE_NAME}")
+    yield (sql_name, hook)
+    if not isinstance(hook, BigQueryHook):
+        hook.run(f"DROP TABLE IF EXISTS {schema}.{OUTPUT_TABLE_NAME}")
```

Comment:
This code seems to be the same as:
https://github.com/astro-projects/astro/blob/df8492fe3d40574cba72e56f7ab23f6725ad923d/tests/operators/test_agnostic_load_file.py#L473-L486

@dimberman what do you think of refactoring this, so both tests can share the same fixture?

Reply:
@tatiana great call, now both tests use the same code